controllers/s3utils.go

```diff
@@ -365,7 +368,10 @@ func (s *s3ObjectStore) UploadObject(key string,
 			bucket, key, err)
 	}
 
-	if _, err := s.uploader.Upload(&s3manager.UploadInput{
+	ctx, cancel := context.WithDeadline(context.TODO(), time.Now().Add(uploadTimeout))
+	defer cancel()
+
+	if _, err := s.uploader.UploadWithContext(ctx, &s3manager.UploadInput{
```

Comment:
Will this return an awsErr? As that is the trigger in `PVUploadToObjectStore` to assume it is a persistent err and not attempt to upload other PVs in the current reconcile.

If the error would be a generic context deadline exceeded then the error caching will be subverted.

The awsErr when s3 is down has been noted at 2 minutes, and the deadline is 30s, hence this may slow down failover taking 30s per PVC (so 10 minutes for 20 PVCs) as compared to the cached error that would take a 2 minute delay on the first upload and then ~0s on the rest.

Reply:
Good point. Yes, it will return a generic context deadline exceeded error.

I will update the patch such that the timeout is greater than 2 mins, just to give the s3 store to fail with the awsErr. In addition to that, I will also update the error caching code to interpret the timeout errors similar to awsErr.

Reply:
I checked and I see that the errors are awsErr types. No extra work is needed.